src/legacy/core_plugins/metrics/server/lib/search_strategies/searh_requests/search_request.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import AbstractSearchRequest from './abstract_request';
+
+import MultiSearchRequest from './multi_search_request';
+import SingleSearchRequest from './single_search_request';
+
+export default class SearchRequest extends AbstractSearchRequest {
+  getSearchRequestType(options) {
+    const isMultiSearch = Array.isArray(options.body);
+    const SearchRequest = isMultiSearch ? MultiSearchRequest : SingleSearchRequest;
```

Comment:
Do we really need to make that distinction? Couldn't we just use `msearch` also for the cases we get in a single body?

Reply:
Are you sure we need to fix it? 'search' works faster than 'msearch' for single queries.  But you are right we can use msearh for all cases. @timroes  should we fix that?